File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void){

    string text;
    int n;

    do{
        text = get_string("type in text: ");
        n = strlen(text);
    }
    while (!(text[n-1] == '!' || text[n-1] == '?' || text[n-1] == '.'));

    float letters = count_letters(text);
    float words = count_words(text);
    float sentenses = count_sentences(text);

//Coleman-Liau index variables

    float L = (letters/words) * 100;
    float S = (sentenses/words) * 100;

// Coleman-Liau algorithm.

    double index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);

    if (grade < 1){

        printf("Before Grade 1\n");
   }
   else if (grade >= 16){
        printf("Grade 16+\n");
   }
   else {
    printf("Grade %i\n", grade);
   }

}
//letters counter
int count_letters(string text){

    int letter = 0;

    for (int i = 0, n = strlen(text); i <= n; i++){

        if (isalpha(text[i])){
            letter++;
        }
    }
    return letter;

}
// words counter
int count_words(string text){

    int word = 0;

// word assume
    if (text[0] != '\0' && text[0] != 32){

        for (int i = 0 , n = strlen(text); i <= n; i++){
            if (text[i] == 32){
            word++;
            }
        }
        return word + 1;
    }
    else {
    return 0;
    }
}

// sentences counter
int count_sentences(string text){

    int sentens = 0;

    for (int i = 0 , n = strlen(text); i <= n; i++){

         if (text[i] == 46 || text[i] == 63 || text[i] == 33){
            sentens++;
         }
    }
     return sentens;
}

File: sentimental-readability/readability.py
import re

# prompt user for string
while True:
    text = input("Text: ")
    if len(text) == 0:
        continue
    else:
        break

# letters counter
def letters_counter(text):
    letters = 0
    for i in range(len(text)):
        x = text[i].isalpha()
        if x:
            letters = letters + 1

    return letters

# word counter
def word_counter(text):
    words = 0
    for i in range(len(text)):
        x = text[i].isspace()
        if x:
            words = words + 1

    return words + 1

# sentences counter
def sentences_counter(text):
    sentences = 0
    for i in range(len(text)):
        x = re.match("[/.!?]", text[i])
        if x:
            sentences = sentences + 1

    return sentences

# Coleman-Liau index
def Coleman_Liau_index(L,S):


    return round(0.0588 * L - 0.296 * S - 15.8)

# set variables for Coleman_Liau_index
L = (letters_counter(text) / word_counter(text)) * 100
S = (sentences_counter(text) / word_counter(text)) * 100

result = Coleman_Liau_index(L,S)

#printing results
if result < 1